nncf/common/engine.py

```diff
@@ -13,73 +13,22 @@
 
 from abc import ABC
 from abc import abstractmethod
-from typing import Callable, TypeVar, Dict
-
-
-from nncf.common.tensor import NNCFTensor
-from nncf.common.tensor_statistics.statistic_point import StatisticPointsContainer
+from typing import TypeVar, Any
 
 TModel = TypeVar('TModel')
-ModelInput = TypeVar('ModelInput')
+
 
 class Engine(ABC):
     """
     The basic class aims to provide the interface to infer the model.
     """
 
-    def __init__(self):
-        self.model = None
-        self._dataset = None
-        self._inputs_transforms = lambda input_data: input_data
-        self._outputs_transforms = lambda output_data: output_data
-
-    # TODO (Nikita Malinin): Add statistic aggregator object (per-backend)
-    @property
-    def dataset(self):
-        return self._dataset
-
-    def set_dataset(self, dataset) -> None:
-        self._dataset = dataset
-
-    def set_model(self, model: TModel) -> None:
-        self.model = model
-
-    def is_model_set(self) -> bool:
-        return self.model is not None
-
-    def set_outputs_transforms(self, outputs_transforms: Callable):
-        """
-        Sets outputs transforms that applies to the model outputs after inference.
-        """
-        self._outputs_transforms = outputs_transforms
-
-    def set_inputs_transforms(self, inputs_transforms: Callable):
-        """
-        Sets inputs transforms that applies to the input before inference.
-        """
-        self._inputs_transforms = inputs_transforms
-
     @abstractmethod
-    def compute_statistics(self, statistic_points: StatisticPointsContainer, subset_size: int = None) -> None:
-        """
-        Performs model inference on specified dataset subset and collects statistics
-
-        :param statistic_points: StatisticPointsContaine with StatisticPoints,
-         in which statistics are collected and registered.
+    def infer(self, input_data: Any) -> Any:
```

Comment:
```suggestion
    def infer(self, input: Any) -> Any:
```
Are you going to handle the tuple input argument in the infer method? If yes, what do you think about implement it in here?

I ask because the transformation function transforms a data item into model input as the following:
1. A model has one input argument:
```
for data_item in val_loader:
    model(transform_fn(data_item))
```
2. A model has multiple input arguments:
```
for data_item in val_loader:
    args = transform_fn(data_item)
    model(*args)
```

Reply:
Need to file a task to support models with multiple input arguments.